pkg/operator/configobserver/node/observe_latency_profile.go

```diff
@@ -0,0 +1,202 @@
+}
+
+// NewSuppressConfigUpdateUntilSameProfileFunc is used to create a conditional func (shouldSuppressConfigUpdatesFunc)
+// that can be used by the latency profile config observer to determine if a new update to observedConfig should be rolled out or not.
+// It uses a config map lister and status obtained from a static pod operator client to check if all active revisions on the cluster have
+// common values for the required latency config paths or not. Config maps generated by installer controller
+// are read in order to determine the current active static pod revision and compare observedConfig values from it.
+func NewSuppressConfigUpdateUntilSameProfileFunc(
+	operatorClient v1helpers.StaticPodOperatorClient,
+	configMapLister listersv1.ConfigMapNamespaceLister,
+	latencyConfigs []LatencyConfigProfileTuple,
+) (f func() (bool, error)) {
+
+	usedConfigPaths := make([][]string, len(latencyConfigs))
+	for i, latencyConfig := range latencyConfigs {
+		usedConfigPaths[i] = latencyConfig.ConfigPath
+	}
+
+	ret := revisionDiffProfileSuppressor{
+		operatorClient:  operatorClient,
+		configMapLister: configMapLister,
+		usedConfigPaths: usedConfigPaths,
+	}
+
+	// creates an actual conditional fn that can be invoked each time config observer
+	// intends to update observed config
+	return ret.shouldSuppressConfigUpdates
+}
+
+func (s *revisionDiffProfileSuppressor) shouldSuppressConfigUpdates() (suppress bool, err error) {
+	_, operatorStatus, _, err := s.operatorClient.GetStaticPodOperatorState()
+	if err != nil {
+		return false, err
+	}
+
+	// Collect the unique set of revisions for all nodes in cluster
+	uniqueRevisionMap := make(map[int32]bool)
+	for _, nodeStatus := range operatorStatus.NodeStatuses {
+		revision := nodeStatus.CurrentRevision
+		uniqueRevisionMap[revision] = true
+	}
+
+	var lastConfig map[string]interface{}
+	for revision := range uniqueRevisionMap {
+		configMapNameWithRevision := fmt.Sprintf("%s-%d", revisionConfigMapName, revision)
+		configMap, err := s.configMapLister.Get(configMapNameWithRevision)
+		if err != nil {
+			return false, err
+		}
+
+		// read observed config from current config map
+		configData, ok := configMap.Data[revisionConfigMapKey]
+		if !ok {
+			return false, fmt.Errorf("could not find %s in %s config map from %s namespace", revisionConfigMapKey, configMap.Name, configMap.Namespace)
+		}
+
+		var currentConfig map[string]interface{}
+		if err := json.Unmarshal([]byte(configData), &currentConfig); err != nil {
+			return false, err
+		}
+
+		// prune the current config with only paths that is supposed to be monitored
+		// and keep comparing with the last config, in case of mis match, suppress
+		currentConfigPruned := configobserver.Pruned(currentConfig, s.usedConfigPaths...)
+		if lastConfig != nil && !reflect.DeepEqual(lastConfig, currentConfigPruned) {
+			// suppress=true: when config values don't match
+			return true, nil
+		}
+		lastConfig = currentConfigPruned
```

Comment:
mhm, I think we should compare against `operatorSpec.ObservedConfig.Raw` besides we should narrow the comparison down only to some specific arguments. Right now we are comparing the entire configuration. Does it make sense?

Reply:
In fact we should merge `operatorSpec.ObservedConfig.Raw` and `operatorSpec.UnsupportedConfigOverrides.Raw` and compare against it.

Reply:
It could be an issue handling `UnsupportedConfigOverrides` in case someone sets the arg on the cluster for the ones which KCM/KApiS use for latency profiles.

> am okay to tell that the operator is trying forever as well... The cluster is in that case in unsupported mode anyway.

This might be helpful: https://github.com/openshift/cluster-kube-controller-manager-operator/pull/611#discussion_r857868140

Reply:
+1 for merging with `UnsupportedConfigOverrides`

Reply:
Yeah, we moving ahead with merging! (added the change)